stdlib/public/runtime/Metadata.cpp

```diff
@@ -3655,6 +3655,17 @@ class ForeignMetadataCacheEntry
         state = PrivateMetadataState::NonTransitiveComplete;
       }
     } else {
+      if (candidate->getValueWitnesses() == nullptr) {
+        assert(isa<ForeignClassMetadata>(candidate) &&
+               "expected a ForeignClassMetadata");
+#if SWIFT_OBJC_INTEROP
+        // FIXME(SR10054): this should really check if the class was built with
+        // ObjC interop enabled.
```

Comment:
I don't think that's necessary; ObjC interop (at least as we deal with it on Darwin) really has to be an ABI-wide setting, not something you can enable in a specific module.

But please add a comment on this whole `if` saying something like "Default the VWT if it wasn't set in on the candidate."  Also, the message in that assertion is useless; please change it to something like "cannot set default value witnesses for non-class foreign types".